### PR TITLE
fix balancer liquidity v2

### PR DESCRIPTION
**Pull request by mendesfabio**
Adds `token_amount_raw` and `token_amount` columns to the `liquidity` views and fixes an issue when joining with token weights for estimating liquidity.

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by Hosuke** [comment]
We may need to fix the duneSQL compatibility in this PR.
https://github.com/duneanalytics/spellbook/actions/runs/4596713919/jobs/8118522392?pr=3039#step:14:18

**Comment by mendesfabio** [comment]
it's not clear to me how to make it compatible with both engines -- do you have any examples? also, do I have to fix it on my side or will you guys do it?

File: models/balancer/arbitrum/balancer_v2_arbitrum_liquidity.sql [hunk 1]
```diff
@@ -20,10 +20,11 @@ WITH pool_labels AS (
         SELECT
             date_trunc('day', minute) AS day,
             contract_address AS token,
+            decimals,
             AVG(price) AS price
         FROM {{ source('prices', 'usd') }}
         WHERE blockchain = "arbitrum"
-        GROUP BY 1, 2
+        GROUP BY 1, 2, 3
```
Comment (Hosuke) [inline models/balancer/arbitrum/balancer_v2_arbitrum_liquidity.sql:27]: We may need to rewrite the group by in full column names.

Comment (Hosuke) [reply models/balancer/arbitrum/balancer_v2_arbitrum_liquidity.sql:27]: Substring function may also varies.

**Review by Hosuke — COMMENTED** [review]
We may try to run the compiled query on dune sql first.
Current codes fits the spark sql syntax, but for dune sql there are some built-in functions need to be replaced..

File: models/balancer/arbitrum/balancer_v2_arbitrum_liquidity.sql (hunk 1 above)
Comment (mendesfabio) [reply models/balancer/arbitrum/balancer_v2_arbitrum_liquidity.sql:27]: should I do this change? how do I make it compatible with both Spark and Dune SQL if I use built-in functions for e.g. generating a time series that is different in each engine?

Comment (mendesfabio) [reply models/balancer/arbitrum/balancer_v2_arbitrum_liquidity.sql:27]: are PRs only merged if the spell is compatible with both engines from now onwards? that's not clear to me too

**Review by Hosuke — APPROVED** [review]
LGTM.✅
This PR introduces `token_balance_raw` and `token_balance` for balancer_v2_*_liquidity.
Thank you @mendesfabio

**Comment by jeff-dude** [comment]
hi @mendesfabio sorry for the confusion on the separate engines in spellbook right now. the issue we have is that the backend ETL pipelines still run on spark, yet the spells created ideally can run on dunesql. views in particular are the most difficult, as it has to succeed on spark and dunesql (trino) syntax. if we don't make it spark compatible, then the ETL process will fail. if we don't make it trino compatible, then querying the view on the app will fail (hence the error in the CI test attached to PR).

for these particular views, the current error message in the gh action CI test is yelling at us for using the `explode` function, as it's not available on trino.
```
    calendar AS (
        SELECT explode(sequence(to_date('2021-04-21'), CURRENT_DATE, interval 1 day)) AS day
    ),
```

we either need to look into materializing these spells into tables, which makes it easier to run on spark/query just the output on dunesql (rather than run the underlying view code each time), or find a function that is compatible for both engines for `explode`

if we explore materializing, how often do you need these updated? if only 1x/day, it can be a standard table. if multiple times a day, it'd need to be an incremental table.

if you can find a function to replace `explode`, then it can remain a view

**Comment by markusbkoch** [comment]
cc @Stefenon @thetroyharris FYI, see Jeff's comment

**Comment by mendesfabio** [comment]
once a day is good enough for now, can you materialize this table @jeff-dude? thanks in advance!

**Review by jeff-dude — APPROVED** [review]
okay, got them materialized as tables. they'll refresh 1x/day

**Pull request merged**